### PR TITLE
add option to configure multiple listening addresses

### PR DESCRIPTION
**Pull request by Garionion**
this PR adds the option to configure multiple addresses directly instead of relying on DNS

However, I discovered a bug in srt (i guess), where I can only listen on one Address for both IPv4 and IPv6 at a time. the second Address of a type fails with ` Error in srt_listen`
but i guess this is not a bug introduced by this PR so … 
EDIT: apparently this error does not occur, if i use different ports … i'm confused

### CODE REVIEW

File: srt/server.go [hunk 1]
```diff
@@ -277,7 +274,8 @@ func (s *ServerImpl) registerForStats(ctx context.Context, conn *srtConn) {
 func (s *ServerImpl) GetStatistics() []*relay.StreamStatistics {
 	streams := s.relay.GetStatistics()
 	for _, stream := range streams {
-		stream.URL = fmt.Sprintf("srt://%s?streamid=play/%s", s.config.Address, stream.Name)
+		//TODO how to decide which Address to use
```
Comment (iSchluff) [inline srt/server.go:277]: I think using the first address should be fine as the user can reorder the addresses accordingly.

File: config.toml.example [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [app]
 # Relay listen address
-#address = "127.0.0.1:1337"
+#addresses = ["127.0.0.1:1337", "[::1]:1337"]
```
Comment (iSchluff) [inline config.toml.example:3]: Please use localhost for the default and add a comment saying that multiple addresses may be specified.

File: config/config.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func Parse(paths []string) (*Config, error) {
 	// set defaults
 	config := Config{
 		App: AppConfig{
-			Address:     "127.0.0.1:1337",
+			Addresses:   []string{"127.0.0.1:1337", "[::1]:1337"},
```
Comment (iSchluff) [inline config/config.go:57]: I would also use localhost in the defaults as the user may have ipv6 disabled

File: srt/server.go [hunk 4]
```diff
@@ -67,31 +66,29 @@ func NewServer(config *Config) Server {
...
-
-	for _, address := range addrs {
-		err := s.listenAt(ctx, address, uint16(port))
+	for _, address := range s.config.Addresses {
+		host, portString, err := net.SplitHostPort(address)
+		if err != nil {
+			return err
+		}
+		if len(host) == 0 {
+			host = "localhost"
```
Comment (iSchluff) [inline srt/server.go:75]: I think that does not match the default behavior in go. Go binds to all addresses when the host is omitted. So I think in this case we should bind to `0.0.0.0` and `::`

**Comment by Garionion** [comment]
Just for the record: I've done all the requested changes, but when i want to listen on all v6 addresses, i get:
```
srtrelay on  master [!?] via  v1.16.5 
❯ ./srtrelay -addresses :1337
2021/07/02 15:16:34 Config file not found, using defaults
2021/07/02 15:16:34 SRT Listening on 0.0.0.0:1337
2021/07/02 15:16:34 Listen failed for [::]:1337 : Error in CreateAddrInet, LookupIP
```
I found this [issue](https://github.com/Haivision/srt/issues/1481) but it does not relate to the problem i have here?

File: srt/server.go [hunk 5]
```diff
@@ -67,31 +66,34 @@ func NewServer(config *Config) Server {
...
+			return err
+		}
+		var addrs []string
+		if len(host) != 0 {
+			addrs, err = net.LookupHost(host)
+			if err != nil {
+				return err
+			}
+		} else {
+			addrs = []string{"0.0.0.0", "[::]"}
```
Comment (iSchluff) [inline srt/server.go:87]: > Just for the record: I've done all the requested changes, but when i want to listen on all v6 addresses, i get:
> 
> ```
> srtrelay on  master [!?] via  v1.16.5 
> ❯ ./srtrelay -addresses :1337
> 2021/07/02 15:16:34 Config file not found, using defaults
> 2021/07/02 15:16:34 SRT Listening on 0.0.0.0:1337
> 2021/07/02 15:16:34 Listen failed for [::]:1337 : Error in CreateAddrInet, LookupIP
> ```

```[::]``` is not a valid IP address here as we expect parsed addresses. Also found out that apparently listening on ```::``` is enough to also receive IPv4 connections.

**Comment by iSchluff** [comment]
Rebased and merged to master. The fix for the issue above is in the latest commit.
Thanks a lot

**Pull request closed without merging**